docs/how-tos/protocol.rst

```diff
@@ -2,3 +2,536 @@
+            description="Length of each simulation"
+        )
+        
+        equilibration_length: pydantic.Field(
+            default=1.0 * unit.nanosecond, 
+            description="Equilibration time before production"
+        )
+
+
+Step 2: Define your ProtocolResult
+----------------------------------
+
+Create a result class that knows how to compute estimates and uncertainties from your protocol's outputs:
+
+.. code-block:: python
+
+    from gufe import ProtocolResult
+    from openff.units import unit
+    import numpy as np
+
+    class MyProtocolResult(ProtocolResult):
+        
+        def get_estimate(self) -> unit.Quantity:
+            """Calculate the free energy estimate from all runs."""
+            # Extract the key results from all completed runs
+            free_energies = []
+            for run_result in self.data["free_energies"]:
+                free_energies.append(run_result)
+            
+            # Return the mean as our best estimate
+            mean_dg = np.mean(free_energies) * unit.kilocalorie_per_mole
+            return mean_dg
+        
+        def get_uncertainty(self) -> unit.Quantity:
+            """Calculate the uncertainty from all runs."""
+            free_energies = self.data["free_energies"]
+            if len(free_energies) < 2:
+                return 0.0 * unit.kilocalorie_per_mole
+            
+            std_err = np.std(free_energies) / np.sqrt(len(free_energies))
+            return std_err * unit.kilocalorie_per_mole
+
+
+Step 3: Define your ProtocolUnits
+----------------------------------
+
+Create the computational units that will perform the actual work.
+Each unit should inherit from :class:`.ProtocolUnit` and implement a ``_execute`` method.
+
+.. important :: 
+
+   Use ``ctx.shared`` for large objects that need to be passed between units.
+   This avoids serialization issues and improves performance by keeping file paths in the return objects instead of the large objects themselves.
+
+.. code-block:: python
+
+    from gufe import ProtocolUnit
+
+    class SetupUnit(ProtocolUnit):
+        """Prepare the system for simulation."""
+        
+        @staticmethod
+        def _execute(ctx, *, stateA, stateB, mapping, settings, **inputs):
+            """Set up the alchemical system."""
+            import pickle
+            from pathlib import Path
+            
+            # ctx provides scratch and shared directories
+            # Use ctx.shared to write files that other units will need
+            shared_dir = Path(ctx.shared)
+            
+            # Your setup logic here...
+            # - Create alchemical system from stateA/stateB
+            # - Apply the atom mapping 
+            # - Set up force field parameters
+            prepared_system = ...  # Your setup code here
+            topology = ...         # Your topology creation
+            coordinates = ...      # Your coordinate preparation
+            
```

Comment:
maybe i'm misreading, but this is a bit confusing to me. this docstring says to set up force field parameters, but where does that happen?

Reply:
How all this happens is going to be very dependent on the tools and methods one is using. I think this is about as far as we can go with recommendations here.